.github/workflows/pr-builder.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will build the project on pull requests with tests
+# Uses:
+#   OS: ubuntu-latest
+#   JDK: Adopt JDK 8
+
+name: PR Builder
+
+on:
+  pull_request:
+    branches: [main, master]
+  workflow_dispatch:
+
+env:
+  MAVEN_OPTS: -Xmx4g -Xms1g
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Adopt JDK 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: "11"
+          distribution: "adopt"
+      - name: Cache local Maven repository
+        id: cache-maven-m2
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-m2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ env.cache-name }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-${{ env.cache-name }}-
+            ${{ runner.os }}-maven-
+            ${{ runner.os }}-
+      - name: Build with Maven
```

Comment:
Delete .m2 after building

Reply:
Let's delete the snapshot artifacts only refer: https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/.github/workflows/pr-builder.yml

Reply:
fixed with https://github.com/wso2-extensions/identity-oauth-dpop/commit/d213e4203f00f4eccfca4db70b07e55a1c4ae5ca